spec/fixtures/site/page.md

```diff
@@ -27,8 +27,12 @@
 
 [A first fragment inline](another-page.md#foo) and [a second fragment in the same line](page-with-permalink.md#bar)
 
+[A indented reference][indented-reference]
+
 [reference]: another-page.md
 
 [reference-with-fragment]: another-page.md#foo
 
 [reference-brackets]: another-page.md#(bar)
+
+  [indented-reference]: another-page.md
```

Comment:
Just to ensure we are testing every edge case and further work doesn't break this, perhaps we should test linking from a page to a document?

Reply:
oops, totally forgot adding tests for those.